Swiften/Serializer/PayloadSerializers/UnitTest/MIXJoinSerializerTest.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#include <gtest/gtest.h>
+
+#include <Swiften/Serializer/PayloadSerializers/MIXJoinSerializer.h>
+
+using namespace Swift;
+
+TEST(MIXJoinSerializerTest, XEP0369_Example22) {
+    MIXJoinSerializer testling;
+
+    std::shared_ptr<MIXJoin> join(new MIXJoin());
```

Comment:
Please use `auto join = std::make_shared<MIXJoin>()` in future. This has less repetition of the typename. For more details see the answer to question 2 on https://herbsutter.com/2013/05/29/gotw-89-solution-smart-pointers/ .